locked-nft/contracts/NFTLocker.cdc

```diff
@@ -0,0 +1,180 @@
+import NonFungibleToken from "./NonFungibleToken.cdc"
+
+pub contract NFTLocker {
+
+    pub event ContractInitialized()
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+    pub event NFTLocked(
+        id: UInt64,
+        to: Address?,
+        lockedAt: UInt64,
+        lockedUntil: UInt64,
+        duration: UInt64,
+        nftType: Type
+    )
+    pub event NFTUnlocked(
+        id: UInt64,
+        from: Address?,
+        nftType: Type
+    )
```

Comment:
Think it might be useful to include a `Display` view or something like that in these?

Reply:
Good question, I thought about it a bit, and this contract does not implement the NFT interface. It stores NFT from multiple collections, but does not emit its own NFT. Would need to call the base contract of the asset and that does not seem possible, afaik. Or could implement a generic display, but not sure where the best place to do that would be given it is not an NFT contract, per se.